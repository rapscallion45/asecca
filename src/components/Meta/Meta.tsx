import { FC } from 'react';
import { useTheme } from '@mui/material';

/**
 * Meta
 *
 * Applications meta data component for filling page Head element,
 * with app name, description, SEO, favicon, etc
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @component
 * @returns {FC} - meta functional component
 */
const Meta: FC = () => {
  /* shorthand theme helper */
  const theme = useTheme();

  return (
    <>
      <title>{process.env.NEXT_PUBLIC_APP_NAME}</title>
      <meta
        name="description"
        content={process.env.NEXT_PUBLIC_APP_DESCRIPTION}
      />
      <meta name="viewport" content="width=device-width, initial-scale=1" />
      <link rel="icon" href="/favicon.ico" />
      <link
        rel="apple-touch-icon"
        sizes="180x180"
        href="/favicon/apple-touch-icon.png"
      />
      <link
        rel="icon"
        type="image/png"
        sizes="32x32"
        href="/favicon/favicon-32x32.png"
      />
      <link
        rel="icon"
        type="image/png"
        sizes="16x16"
        href="/favicon/favicon-16x16.png"
      />
      <link rel="manifest" href="/favicon/site.webmanifest" />
      <link
        rel="mask-icon"
        href="/favicon/safari-pinned-tab.svg"
        color={theme.palette.primary.main}
      />
      <link rel="icon" href="/favicon/favicon.ico" />
      <meta
        name="msapplication-TileColor"
        content={theme.palette.primary.main}
      />
      <meta name="msapplication-config" content="/favicon/browserconfig.xml" />
      <meta
        name="apple-mobile-web-app-title"
        content={process.env.NEXT_PUBLIC_APP_NAME}
      />
      <meta
        name="application-name"
        content={process.env.NEXT_PUBLIC_APP_NAME}
      />
      <meta name="theme-color" content={theme.palette.primary.main} />
      <link rel="alternate" type="application/rss+xml" href="/feed.xml" />
      <link rel="preconnect" href="https://fonts.googleapis.com" />
      <link rel="preconnect" href="https://fonts.gstatic.com" />
    </>
  );
};

export default Meta;
