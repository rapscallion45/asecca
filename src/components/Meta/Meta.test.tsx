import React from 'react';
import renderer from 'react-test-renderer';
import Meta from './Meta';

/* Meta Unit Tests */
/* =============== */
describe('Meta', () => {
  it('Renders correctly', async () => {
    /* perform snapshot test */
    const tree = renderer.create(<Meta />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
