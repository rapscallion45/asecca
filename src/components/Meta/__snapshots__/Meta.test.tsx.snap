// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta Renders correctly 1`] = `
[
  <title>
    Asecca
  </title>,
  <meta
    content="Asecca front end client SPA"
    name="description"
  />,
  <meta
    content="width=device-width, initial-scale=1"
    name="viewport"
  />,
  <link
    href="/favicon.ico"
    rel="icon"
  />,
  <link
    href="/favicon/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />,
  <link
    href="/favicon/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />,
  <link
    href="/favicon/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />,
  <link
    href="/favicon/site.webmanifest"
    rel="manifest"
  />,
  <link
    color="#1976d2"
    href="/favicon/safari-pinned-tab.svg"
    rel="mask-icon"
  />,
  <link
    href="/favicon/favicon.ico"
    rel="icon"
  />,
  <meta
    content="#1976d2"
    name="msapplication-TileColor"
  />,
  <meta
    content="/favicon/browserconfig.xml"
    name="msapplication-config"
  />,
  <meta
    content="Asecca"
    name="apple-mobile-web-app-title"
  />,
  <meta
    content="Asecca"
    name="application-name"
  />,
  <meta
    content="#1976d2"
    name="theme-color"
  />,
  <link
    href="/feed.xml"
    rel="alternate"
    type="application/rss+xml"
  />,
  <link
    href="https://fonts.googleapis.com"
    rel="preconnect"
  />,
  <link
    href="https://fonts.gstatic.com"
    rel="preconnect"
  />,
]
`;
